### PR TITLE
Add support for merging values

### PR DESCRIPTION
**Pull request by rossf7**
Towards https://github.com/giantswarm/giantswarm/issues/4949

Adds support for merging values. This will be used in chart-operator, app-operator and opsctl for merging values when installing or updating Helm releases.

Logic is based on the upstream merge logic in helm. See other comments for the link.

### CODE REVIEW

File: merge_values.go [hunk 1]
```diff
@@ -7,6 +7,93 @@ import (
...
+// byte array. Only a single key is supported and the input data structure
+// matches the configmap or secret where the data is stored.
+func processYAML(inputMap map[string][]byte) (map[string]interface{}, error) {
+	var err error
+
+	result := map[string]interface{}{}
+
+	if len(inputMap) > 1 {
+		return nil, microerror.Maskf(executionFailedError, "merging %d keys is unsupported expected 1 key", len(inputMap))
+	}
```
Comment (rossf7) [inline merge_values.go]: Only a single key with YAML is supported. But the key can have any name so there are no magic strings.

File: merge_values.go [hunk 2]
```diff
@@ -7,6 +7,93 @@ import (
...
+
+	return result, nil
+}
+
+// mergeValues implements the merge logic. It performs a deep merge. If a value
+// is present in both then the source map is preferred.
+//
+// Logic is based on the upstream logic implemented by Helm.
+// https://github.com/helm/helm/blob/240e539cec44e2b746b3541529d41f4ba01e77df/cmd/helm/install.go#L358
```
Comment (rossf7) [inline merge_values.go]: See this link for the upstream code which the merge logic is based on. I just made minor changes to follow our usual conventions.

File: merge_values_test.go [hunk 3]
```diff
@@ -41,13 +41,134 @@ nested:
   deeper:
     bottom: true
     float: 274877.906944
+test: test`
+	overrideNestedYaml = `
```
Comment (xh3b4sd) [inline merge_values_test.go]: What about having separate constant blocks for these things? Took me a while to figure what is going on here.

Comment (rossf7) [reply merge_values_test.go]: Good point that is easier to read. Thanks, done.

**Pull request merged**